### PR TITLE
doc: Candid serialization syntax

### PR DESCRIPTION
**Pull request by crusso**
Finishes off #3232 (which finished off #3155)

In language manual.adoc:
* add syntax
* document semantics

[ ] general, standalone page with simple examples?

### CODE REVIEW

File: doc/modules/language-guide/pages/language-manual.adoc [hunk 1]
```diff
@@ -2300,6 +2300,44 @@ The result of evaluating `<exp> : <typ>` is the result of evaluating `<exp>`.
...
+////
+ea(_) is defined in design doc motoko/design/IDL-Motoko.md, but `encode` and `decode` are not explicitly defined anywhere except in the implementation.
+////
+
+
+NOTE: `from_candid` returns `null` when the argument is a valid Candid encoding of the wrong type. It traps if the blob is not a valid Candid encoding at all.
+
+NOTE: `to_candid` and `from_candid` are syntactic operators, not first-class functions, and must be fully applied in the syntax.
+
+WARNING: the Candid encoding of a value as a blob is not unique and the same value may have many different Candid-representations as a blob.
```
Comment (matthewhammer) [inline doc/modules/language-guide/pages/language-manual.adoc:2337]: No need to hyphenate "Candid-representations".

File: doc/modules/language-guide/pages/language-manual.adoc [hunk 2]
```diff
@@ -2300,6 +2300,44 @@ The result of evaluating `<exp> : <typ>` is the result of evaluating `<exp>`.
...
+For this reason, blobs should never be used to, for instance, compute hashes of values or determine equality, whether across compiler
+versions or even just different programs.
```
Comment (matthewhammer) [inline doc/modules/language-guide/pages/language-manual.adoc:2339]: A good reason for us to include a similarly-generic hashing operator that is unrelated to the Candid encoding as a blob.

**Review by matthewhammer — APPROVED** [review]

File: doc/modules/language-guide/pages/language-manual.adoc (hunk 1 above)
Comment (crusso) [inline doc/modules/language-guide/pages/language-manual.adoc:2337]:
```suggestion
WARNING: the Candid encoding of a value as a blob is not unique and the same value may have many different Candid representations as a blob.
```

File: doc/modules/language-guide/pages/language-manual.adoc (hunk 2 above)
Comment (crusso) [reply doc/modules/language-guide/pages/language-manual.adoc:2339]: yep
